app/controllers/account/news_controller.rb

```diff
@@ -0,0 +1,78 @@
+class Account::NewsController < Account::AccountController
+  rescue_from ActiveRecord::RecordNotFound, with: :not_found
+  rescue_from Pundit::NotAuthorizedError, with: :user_not_authorized
+
+  before_action :set_news_by_role
```

Comment:
isn't this used only in index action?

Reply:
no, its used for index and set_post for other actions